script/bootstrap

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env php
+# script/bootstrap: Resolve all dependencies that the application requires to
+#                   run.
+<?php
+error_reporting(E_ALL);
+
+// Go to LORIS root.
+chdir(dirname(__FILE__) . '/..');
+require('tools/PHP_CLI_Helper.class.inc');
+
+//TODO: Update these values as time passes 
+$required_major_php = 7;
+$required_minor_php = 2;
+$required_major_apache = 2;
+$required_minor_apache = 4;
+// PHP version required for LORIS.
+$required_php = "$required_major_php.$required_minor_php";
+
+/* Validate apache */
+// Get string representation of apache version number
+$apache_parts = explode(
+    '/', 
+    // this command yields e.g. Apache/2.4.34
+    shell_exec(
+        "apache2 -v | " . 
+        "head -n 1 | " .
+        "cut -d ' ' -f 3"
+    )
+);
+$apache_version = end($apache_parts);
+/* Look for the string "$major.$minor" in info string. Also match on versions
+ * higher than minor version because we want AT LEAST that version.
+ */
+$pattern = "/$required_major_apache\.[$required_minor_apache-9]/";
+if (preg_match($pattern, $apache_version) === 0) {
+    $required_apache = "$required_major_apache.$required_minor_apache";
+    die(
+        "ERROR: LORIS requires Apache v$required_apache or higher."
+        . PHP_EOL
+        . "This must be done manually as it has possible security ramifications."
+        . PHP_EOL
+    );
+}
+
+// NOTE Update as time passes.
+// Dependencies last updated for version: 20.0.1
+$apt_dependencies = [
+                       "wget",
+                       "zip",
+                       "unzip",
+                       "php-json",
+                       "npm",
+                       "software-properties-common",
+                       "php-ast",
+                       "php$required_php",
+                       "php$required_php-mysql",
+                       "php$required_php-xml",
+                       "php$required_php-json",
+                       "php$required_php-mbstring",
+                       "php$required_php-gd",
+                       "libapache2-mod-php$required_php",
+                      ];
+
+if (
+    !(installMissingRequirements($apt_dependencies))
+    && (installAptPackages($apt_dependencies))
+) {
+    die(
+        'Could not upgrade all required packages. Exiting.'
+        . PHP_EOL
+    );
+}
+
+
+// Run package managers and add dev flag if supplied to the script
+if (runPackageManagers(isset($argv[1]) && $argv[1] === 'dev')) {
+    echo '[**] Dependencies up-to-date.' . PHP_EOL;
+}
+
+/**
+ * Prints a list of missing apt packages and prompts user to install them.
+ *
+ * @param array $requirements Required packages determined missing earlier
```

Comment:
since other lines end with a dot, this line should also end with a dot
same of other doc block